control/pid_controller_dp/include/pid_controller_dp/pid_controller.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef PID_CONTROLLER_HPP
+#define PID_CONTROLLER_HPP
+
+#include "pid_controller_dp/eigen_typedefs.hpp"
+
+class PIDController {
+   public:
+    explicit PIDController();
+
+    // @brief Calculate the control input tau
+    // @param eta: 7D vector containing the vehicle pose [x, y, z, w, x, y, z]
+    // @param eta_d: 7D vector containing the desired vehicle pose [x, y, z, w,
+    // x, y, z]
```

Comment:
consider using a different notation for quat, since having two sets of xyz in the same vector can be misleading/confusing.

Reply:
Also see my comment relating to quaternion operations, `Eigen::Quaterniond` exists for a good reason :fire: